staging/src/k8s.io/apiserver/pkg/server/connectivity_service.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"bufio"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"k8s.io/klog"
+	"net"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/apiserver/pkg/apis/apiserver"
+)
+
+var directDialer utilnet.DialFunc = (&net.Dialer{
+	Timeout:   30 * time.Second,
+	KeepAlive: 30 * time.Second,
+	DualStack: true,
+}).DialContext
+
+type ConnectivityService struct {
+	m  map[string]utilnet.DialFunc
+	mu sync.Mutex
+}
+
+type NetworkContext struct {
+	// ConnectivityServiceName is the unique name of the
+	// ConnectivityServiceConfiguration which determines
+	// the network we route the traffic to.
+	ConnectivityServiceName string
+}
+
+type ConnectivityServiceLookup func(networkContext NetworkContext) (utilnet.DialFunc, error)
+
+func createConnectDialer(connection apiserver.Connection) (utilnet.DialFunc, error) {
```

Comment:
@cheftako I'd like eyes from networking and auth on this section. I'm sure you've got them lined up, but I'd like someone more knowledgable inspecting here.

Reply:
Agreed.

Reply:
Just a reminder.  There may have been nothing to say.  Just looking an affirmative, "they've checked"